general/g.region/main.c

```diff
@@ -828,14 +854,15 @@ int main(int argc, char *argv[])
 	
     /* save= */
     if ((name = parm.save->answer)) {
+	update_file = false;
 	temp_window = window;
 	G_adjust_Cell_head3(&temp_window, 0, 0, 0);
 	if (G_put_element_window(&temp_window, "windows", name) < 0)
 	    G_fatal_error(_("Unable to set region <%s>"), name);
     }
 
     G_adjust_Cell_head3(&window, row_flag, col_flag, 0);
-    if (set_flag) {
+    if (flag.force->answer || (update_file && !flag.noupdate->answer)) {
```

Comment:
I'm not 100% sure I understand the logic here: if any of the conditions that set update_file to true are set and -u is not set, then the file will be updated, whatever the force flag ? Why do we need a force flag then, especially since it is mutually exclusive to noupdate ?

I seem to be misunderstanding something...

Reply:
The update_file variable determines the default behavior, so it is true only if options like res=, raster=, etc are used, otherwise it will be false. But there are cases when you want to enforce different behavior. For example you want to only print modified region but not actually update it: `g.region res=1 -up`
Alternatively, and this would be probably rare, you want to update the region although the new default behavior wouldn't update it, e.g. `g.region res=1 -o save=myregion` (you want to save the region into a named region and update the current one).

Reply:
> I'm not 100% sure I understand the logic here: if any of the conditions that set update_file to true are set and -u is not set, then the file will be updated, whatever the force flag ? Why do we need a force flag then, especially since it is mutually exclusive to noupdate ?
> 
> I seem to be misunderstanding something...

I think this is correct: with "Force update" flag even a `g.region -p -o` writes out the file, which `g.region -p` wouldn't; on the other hand e.g. `g.region vector=lakes -u` wouldn't make much sense (hence trigger error).

Reply:
I added parser rule to avoid cases you mentioned (`g.region vector=lakes -u`) because they can make no sense and could be misunderstood.